<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1233</width>
    <height>795</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QTabWidget" name="regex">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="re">
       <attribute name="title">
        <string>正则表达式</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>正则表达式</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QTextEdit" name="re_text"/>
          </item>
          <item>
           <widget class="QPushButton" name="openreButton">
            <property name="text">
             <string>打开正则表达式文件</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="generate_reButton">
            <property name="text">
             <string>开始生成</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="savere_Button">
          <property name="text">
           <string>保存正则表达式</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="NFA">
       <attribute name="title">
        <string>生成NFA</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>NFA状态转移表</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="NFA_tableWidget"/>
        </item>
        <item>
         <widget class="QLineEdit" name="NFA_start"/>
        </item>
        <item>
         <widget class="QLineEdit" name="NFA_end"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="DFA">
       <attribute name="title">
        <string>生成DFA</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>DFA状态转移表</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="DFA_tableWidget"/>
        </item>
        <item>
         <widget class="QLineEdit" name="DFA_start"/>
        </item>
        <item>
         <widget class="QLineEdit" name="DFA_end"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="minimize">
       <attribute name="title">
        <string>最小化DFA</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>最小化DFA状态转移表</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="MIN_tableWidget"/>
        </item>
        <item>
         <widget class="QLineEdit" name="MIN_start"/>
        </item>
        <item>
         <widget class="QLineEdit" name="MIN_end"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="LEX">
       <attribute name="title">
        <string>生成词法分析程序</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>词法分析程序</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="code_textEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="savecode_Button">
          <property name="text">
           <string>保存程序文件</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="ANA">
       <attribute name="title">
        <string>使用分析程序分析</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>分析程序/分析结果</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="ana_textEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="opencode_Button">
          <property name="text">
           <string>打开被分析代码文件</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="openlex_Button">
          <property name="text">
           <string>打开分析程序</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="startana_Button">
          <property name="text">
           <string>开始分析</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="savelex_Button">
          <property name="text">
           <string>保存分析结果</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="LALR1t">
      <property name="currentIndex">
       <number>6</number>
      </property>
      <widget class="QWidget" name="grammarr">
       <attribute name="title">
        <string>文法</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="4" column="0">
         <widget class="QPushButton" name="generateButton">
          <property name="text">
           <string>生成DFA以及分析表</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QTextEdit" name="graEdit"/>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>打开文法文件</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="saveButton">
          <property name="text">
           <string>保存文法文件</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>文法文本</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>FIRST</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="0">
         <widget class="QTextEdit" name="firstEdit"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>first集合</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>FOLLOW</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="1" column="0">
         <widget class="QTextEdit" name="followEdit"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>folllow集合</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="LR1DFA">
       <attribute name="title">
        <string>LR(1)DFA</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0">
         <widget class="QTableWidget" name="LR1"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>LR（1）DFA状态转移表</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="LALR1DFA">
       <attribute name="title">
        <string>LALR(1)DFA</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="0">
         <widget class="QTableWidget" name="LALR1"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>LALR（1）DFA状态转移表</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="LALR1TABLE">
       <attribute name="title">
        <string>LALR(1)分析表</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="1">
         <widget class="QTableWidget" name="LALR1table"/>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>LALR（1）分析表</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>用分析表进行文法分析</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>词法分析窗口/文法分析过程</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="a_code_Edit">
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Microsoft YaHei UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;请先使用左侧窗口进行词法分析，再进行语法分析&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="gettoken_Button">
            <property name="text">
             <string>打开词法分析文件</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>打开语义运算符文件</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>语法树</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTreeView" name="a_treeView"/>
          </item>
          <item>
           <widget class="QPushButton" name="gentree_Button">
            <property name="text">
             <string>生成文法分析过程及语法树</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1233</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu20222131051">
    <property name="title">
     <string>20222131051</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>蒋昕玮</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>编译原理项目</string>
    </property>
   </widget>
   <addaction name="menu20222131051"/>
   <addaction name="menu"/>
   <addaction name="menu_4"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
